charmtools/build/tactics.py

```diff
@@ -353,11 +387,34 @@ def trigger(cls, relpath):
         return relpath in cls.FILENAMES
 
     def read(self):
-        self._raw_data = self.load(self.entity.open())
+        if not self._read:
+            super(LayerYAML, self).read()
+            self.schema['properties'] = {
+                self.current.name: {
+                    'type': 'object',
+                    'properties': self.data.pop('defines', {}),
+                    'default': {},
+                },
+            }
+
+    def combine(self, existing):
+        super(LayerYAML, self).combine(existing)
+        self.schema = utils.deepmerge(existing.schema, self.schema)
+        return self
+
+    def lint(self):
+        if 'options' not in self.data:
+            return True
+        validator = extend_with_default(jsonschema.Draft4Validator)(self.schema)
```

Comment:
It looks like in this case 'lint' is preforming a now needed mutation of the object which is a bit odd conceptually in that other tactics would be expected to do something similar. Can we do this in 'read' which should be related to loading the version of that data we expect to use in processing (maybe read should be 'load' or 'process' or 'parse' but that is another papercut).